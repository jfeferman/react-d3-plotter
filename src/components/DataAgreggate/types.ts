export type DataAgreggateProps = {
  data: Array<Array<number>>;
  xScale: Function;
  yScale: Function;
}
