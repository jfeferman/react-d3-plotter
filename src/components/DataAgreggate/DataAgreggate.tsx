import { DataPoint } from '../DataPoint'

export type DataAgreggateProps = {
  data: number[]
  xScale: () => void
  yScale: () => void
  radius?: number
}


export const DataAgreggate = ({data, xScale, yScale, radius}: DataAgreggateProps) => {
  // Create an individual data point
  const createDataPoint = (datum: Array<number>, index: number) => {
    return (
      <DataPoint
        key={index}
        xScale={xScale}
        yScale={yScale}
        coords={datum}
        radius={radius}
      />
    )
  }

  // Do not render any data points when data is undefined or length zero
  if (data && data.length === 0) {
    return null
  }

  // Render the data point agreggate
  return (
    <g>
      {data.map((d, i) => {
        return createDataPoint(d, i)
      })}
    </g>
  )
}
