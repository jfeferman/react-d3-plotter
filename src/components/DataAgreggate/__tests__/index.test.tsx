import React from 'react'
import * as TestRenderer from 'react-test-renderer';
import {
  mockData,
  getXScale,
  getYScale,
} from '../../../utils/testUtil'
import DataAgreggate from '..'

describe('components/DataAgreggate', () => {
  let renderer: TestRenderer.ReactTestRenderer;
  let instance: TestRenderer.ReactTestInstance;

  beforeAll(() => {
    renderer = TestRenderer.create(
      <DataAgreggate
        data={mockData}
        xScale={getXScale()}
        yScale={getYScale()}
      />,
    )
    instance = renderer.root;
  })

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it('verifies there are two circules', () => {
    const dataAgreggate = instance.findByType('g')
    expect(dataAgreggate.props.children).toHaveLength(2)
  });

  it('verifies there are no children when no data is passed', () => {
    const renderer = TestRenderer.create(
      <DataAgreggate
        data={[]}
        xScale={getXScale()}
        yScale={getYScale()}
      />,
    )
    expect(renderer.toJSON()).toMatchSnapshot()
  });
})
