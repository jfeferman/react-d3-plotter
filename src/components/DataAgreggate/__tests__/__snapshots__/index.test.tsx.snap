// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DataAgreggate matches the snapshot of the component 1`] = `
<g>
  <circle
    cx={480}
    cy={60}
  />
  <circle
    cx={68}
    cy={334}
  />
</g>
`;

exports[`components/DataAgreggate verifies there are no children when no data is passed 1`] = `null`;
